pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -1222,6 +1222,13 @@ func applyUpstreamTLSSettings(opts *buildClusterOpts, tls *networking.TLSSetting
 			}
 		}
 	}
+
+	// disable transport socket when cluster type is `Cluster_ORIGINAL_DST` and mtls is autoDetected
+	if cluster.LbPolicy == apiv2.Cluster_CLUSTER_PROVIDED {
```

Comment:
You have check `cluster.GetType() == apiv2.Cluster_ORIGINAL_DST` for original DST here? Other clusters such as cluster used by dynamic proxy can have `Cluster_CLUSTER_PROVIDED`  as LbPolicy.

Reply:
What's type of cluster can have `Cluster_CLUSTER_PROVIDED ` as well?

Reply:
I think the following two are examples that use `CLUSTER_PROVIDED` currently. 

https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/dynamic_forward_proxy_filter.html?highlight=dynamic%20forward%20proxy#dynamic-forward-proxy - Dynamic Forward Proxy Cluster is one example

https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/aggregate_cluster.html?highlight=cluster_provided#example

Reply:
Good info, though in istio we donot generate such cluster yet

Reply:
We do not now but when we add this can cause problem and also I think it safe and right way to check `cluster.GetType()` to see if it is Original DST or not

Reply:
+1. Adding check ORIG_DST clsuter  this more hermetic and future proof.